core/sup_tree_core/version_upgrade_task_queue.ex

```diff
@@ -14,48 +14,62 @@ defmodule AntikytheraCore.VersionUpgradeTaskQueue do
   use GenServer
   alias Antikythera.GearName
   alias AntikytheraCore.Version
+  require AntikytheraCore.Logger, as: L
 
   def start_link() do
     GenServer.start_link(__MODULE__, :ok, [name: __MODULE__])
   end
 
   @impl true
   def init(:ok) do
-    {:ok, %{queue: :queue.new(), task_pid: nil}}
+    {:ok, %{queue: :queue.new(), task_pid: nil, enabled?: true}}
   end
 
   @impl true
   def handle_cast(message, state) do
-    new_state = enqueue_message(state, message) |> run_task_if_possible()
+    new_state =
+      case message do
+        {:upgrade    , instruction} -> enqueue_message(state, instruction) |> run_task_if_possible()
+        {:set_enabled, enabled?   } -> set_enabled(state, enabled?)
+      end
     {:noreply, new_state}
   end
 
   defp enqueue_message(%{queue: q} = state, message) do
     %{state | queue: :queue.in(message, q)}
```

Comment:
Now we have the word "instruction" to represent the specific notion in this `GenServer`, let's make identifiers consistent with the term:
- function name: `enqueue_message` => `enqueue_instruction`
- parameter name: `message` => `instruction`